code/parachain/frame/composable-traits/src/staking/mod.rs

```diff
@@ -196,6 +199,9 @@ pub enum RewardPoolConfiguration<
 
 		// Asset ID (collection ID) of the financial NFTs issued for staking positions of this pool
 		financial_nft_asset_id: AssetId,
+
+		// Minimum amount to be staked
+		minimum_staking_amount: Balance,
```

Comment:
I'm wondering if we could/should make this optional? If a pool doesn't want to set a minimum staked amount, it would just default to the requiring `slash(staked amount) >= ED`

Reply:
but if a pool wants to have the minimum stake larger than that, then it could be configured as such

Reply:
I think this would be a good change in terms of flexibility. Better to plan for it now then need to change it later
+1

Reply:
the number of staked positions that can be created by an attacker(or just an annoying user) increases for the same amount when min stake is low.

Reply:
perhaps we should consider limiting the number of times a user can stake then?

Reply:
That's not going to stop the storage growth for the system since a user can create any number of accounts they want

Reply:
Then should we limit the amount of stakes per pool then?

Reply:
Also, a minimum amount will be different per asset - even if we  set it to something arbitrarily high, any bad actor with enough funds could blow up our storage. (also 10k of one asset is different than 10k of another asset, so we can't just pass a maximum through the Config either)

Reply:
That's going to limit the number of staking opportunities for a token holder, say a PICA holder. Ultimately the limit we want to enforce is the number of staked positions possible per supply of the token staked.

Reply:
>a minimum amount will be different per asset

This is why we are making it a configurable parameter instead of hard coding. We will decide these at the time of the launch ideally together with other params.

Reply:
Which works for pools created by governance, but if/when we allow pools to be created  by anyone, we will need to revisit this (perhaps enforcing `minimum stake > X % of total_issuance`, or something similar)

Reply:
Agree, these are economic simulation level stuff, that we haven't got to yet. Ideally with Pablo launch we can decide on token prices better. But until that time any thing is just relative and subject to human judgement

Reply:
Even Polkadot adjusts this value routinely